src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -534,4 +540,104 @@ private void preprocessACLs(final Query<Component> query, final String inputFilt
             query.setFilter(inputFilter);
         }
     }
+
+    public List<Component> getDependencyGraphForComponent(Project project, Component component) {
+        project.setDependencyGraph(new ArrayList<>());
+        if (project.getDirectDependencies() == null || project.getDirectDependencies().isBlank()) {
+            return project.getDependencyGraph();
+        }
+        String queryUuid = "\".*" + component.getUuid().toString() + ".*\"";
+        final Query<Component> query = pm.newQuery(Component.class, "project == :project && directDependencies.matches(" + queryUuid + ")");
```

Comment:
Since this is a UUID stringified i think it's a false positive. but nonetheless it's not best practice. Does there exist some kind of list parameter in the used persistence manager? @nscuro